server/src/com/thoughtworks/go/config/update/DeletePackageRepositoryCommand.java

```diff
@@ -73,9 +91,12 @@ public PackageRepository getPreprocessedEntityConfig() {
 
     @Override
     public boolean canContinue(CruiseConfig cruiseConfig) {
-        if (!goConfigService.isAdministrator(username.getUsername())) {
-            Localizable noPermission = LocalizedMessage.string("UNAUTHORIZED_TO_OPERATE");
-            result.unauthorized(noPermission, HealthStateType.unauthorised());
+        return isAuthorized();
+    }
+
+    private boolean isAuthorized() {
```

Comment:
This does not extend the superclass that CreateRepo and UpdateRepo do, because the validations and canContinue methods were different. We decided to leave this as is, remember?